Fix build failing with gcc 12.2.1

The Alpine 3.17 container-image based build (using gcc 12) showed this build error 

/home/ubuntu/client_home/workspace/build-21-pr-validation-linux_alpine_x86_64/SapMachine/src/hotspot/os/linux/os_linux.cpp:2261:49: error: the address of 'line' will never be NULL [-Werror=address]
 2261 |     while (keywords_to_match[i] != NULL && line != NULL) {
      |                                                 ^
/home/ubuntu/client_home/workspace/build-21-pr-validation-linux_alpine_x86_64/SapMachine/src/hotspot/os/linux/os_linux.cpp:2253:8: note: 'line' declared here
 2253 |   char line[500];
      |        ^~~~
cc1plus: all warnings being treated as errors
 
This is related to a not needed NULL-check in our SapMachine only coding.

--- a/src/hotspot/os/linux/os_linux.cpp
+++ b/src/hotspot/os/linux/os_linux.cpp
@@ -2258,7 +2258,7 @@ static int check_matching_lines_from_file(const char* filename, const char* keyw
 
   while (fgets(line, sizeof(line), fp) != NULL) {
     int i = 0;
-    while (keywords_to_match[i] != NULL && line != NULL) {
+    while (keywords_to_match[i] != NULL) {
       if (strstr(line, keywords_to_match[i]) != NULL) {
         fclose(fp);
         return i;

